lxd/storage_zfs.go

```diff
@@ -527,11 +529,13 @@ func (s *storageZfs) ContainerCreate(container container) error {
 	containerPath := container.Path()
 	containerName := container.Name()
 	fs := fmt.Sprintf("containers/%s", containerName)
+	dataset := fmt.Sprintf("%s/containers/%s", s.pool.Name, containerName)
```

Comment:
So `s.pool.Name` is not correct as it only holds the name of the storage pool in the db not the on-disk name. You need to use `s.getOnDiskPoolName()`.